tests/e2e/wp/approve.spec.ts

```diff
@@ -18,42 +24,41 @@ test("Admin user can approve a Work Plan submission", async ({ page }) => {
   await createSubmittedWorkPlan(currentYear, firstPeriod);
 
   // View WPs
-  await page
-    .getByRole("combobox", {
-      name: "List of states, including District of Columbia and Puerto Rico",
-    })
-    .selectOption(stateAbbreviation);
-  await page.getByRole("radio", { name: "MFP Work Plan" }).click();
-  await page.getByRole("button", { name: "Go to Report Dashboard" }).click();
-  await expect(page).toHaveURL("/wp");
-  await page.waitForResponse(`**/reports/WP/${stateAbbreviation}`);
+  await adminHomePage.dropdown.selectOption(stateAbbreviation);
+  await adminHomePage.selectWP();
+  await adminHomePage.goToDashboard();
+  await wpDashboard.isReady();
+  await expect(wpDashboard.page).toHaveURL(wpDashboard.path);
```

Comment:
So I started to investigate doing this and the challenge with moving the URL assertion into the `isReady()` function is how to return a "compound" expect from that function? Since we are already returning an assertion that the table has loaded. 

Maybe what is confusing here is the order that I put these calls, _and_ the fact that I am returning an assertion in that `isReady()` function. Brainstorming a better approach:

1. await the click to go to the Dashboard
2. assert that the URL is what we expect
3. remove the `expect` from `isReady()` and instead just await the table header text to be visible

...thoughts?

Reply:
🤔 hmm item 3 would remove expects from our page object models, while still accomplishing what we wanted 
I think this is a good approach

Reply:
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExa2U1bHNvbnJwcGVpemUxN2p4b3NoM3pleGd5am50b3V4aHBqaWxwcSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/curhc38xMnMU8/giphy.gif)